http_check/datadog_checks/http_check/http_check.py

```diff
@@ -115,7 +116,7 @@ def send_status_down(loginfo, down_msg):
             self.http.session.trust_env = False
 
             # Add 'Content-Type' for non GET requests when they have not been specified in custom headers
-            if method.upper() in DATA_METHODS and not headers.get("Content-Type"):
+            if method.upper() in DATA_METHODS and not CaseInsensitiveDict(headers).get("Content-Type"):
```

Comment:
Let's save this type during the config parsing

Reply:
Good idea! This has now been updated